public/pages/CreateSnapshotPolicy/containers/CreateSnapshotPolicy/CreateSnapshotPolicy.tsx

```diff
@@ -674,17 +700,17 @@ export default class CreateSnapshotPolicy extends Component<CreateSMPolicyProps,
             <EuiSpacer size="s" />
 
             <EuiCheckbox
-              id="partial"
-              label="When a snapshot operation failed."
-              checked={notifyOnFailure}
+              id="notification-failure"
+              label="When a snapshot has failed."
+              checked={getNotifyFailure(policy)}
               onChange={(e: ChangeEvent<HTMLInputElement>) => {
                 this.setState({ policy: this.setPolicyHelper("notification.conditions.failure", e.target.checked) });
               }}
             />
           </div>
           {showNotificationChannel ? (
-            <ChannelNotification
-              channelId=""
+            <Notification
+              channelId={_.get(policy, "notification.channel.id") || ""}
```

Comment:
I see you used `??` on [line 623](https://github.com/opensearch-project/index-management-dashboards-plugin/pull/208/files#diff-55ba33382eb0acb4d851705cc07cb1fd0f97f6a62a67d2877a51a28f20ed0815R623). Are the `||` and `??` functioning the same in these cases?

Reply:
They are same when checking null